### PR TITLE
🚸 Enable search text to update creator/tag options

### PR DESCRIPTION
**Pull request by AuroraHuang22**
- Creator

https://github.com/likecoin/liker-land/assets/75730405/d4e77e9c-48cb-4d23-b36c-56d90fac0470

-Keywords

https://github.com/likecoin/liker-land/assets/75730405/d54ebdb3-467d-45ee-9728-ee41f6404ffd

### CODE REVIEW

File: src/components/NFTPortfolio/FilterDropdown.vue [hunk 1]
```diff
@@ -160,7 +180,20 @@ export default {
...
+    return {
+      filterInputCreator: '',
+      filterInputKeywords: '',
+    };
+  },
+  watch: {
+    filterInputCreator(creator) {
+      this.$emit('input-filter-change-creator', creator);
+    },
+    filterInputKeywords(keywords) {
```
Comment (williamchong) [inline src/components/NFTPortfolio/FilterDropdown.vue:193]: Can we use @input instead of watcher?

File: src/locales/zh-Hant.json [hunk 2]
```diff
@@ -437,9 +437,10 @@
   "ok": "OK",
   "order_menu_by": "按{order}",
   "order_menu_collected": "擁有",
-  "order_menu_display_state": "精選",
   "order_menu_filter_by": "篩選",
+  "order_menu_display_state": "精選",
   "order_menu_filter_by_creators": "作者",
+  "order_menu_filter_by_input": "輸入篩選內容",
```
Comment (williamchong) [inline src/locales/zh-Hant.json:443]: 輸入搜尋內容 seems easier to understand

File: src/locales/en.json [hunk 3]
```diff
@@ -450,6 +450,7 @@
   "order_menu_display_state": "Featured",
   "order_menu_filter_by": "Filter",
   "order_menu_filter_by_creators": "Creators",
+  "order_menu_filter_by_input": "Enter filtering content",
```
Comment (williamchong) [inline src/locales/en.json:453]:
```suggestion
  "order_menu_filter_by_input": "Enter search keyword",

```

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/mixins/portfolio.js [hunk 4]
```diff
@@ -246,15 +248,26 @@ export const createPortfolioMixin = ({
...
+                id.includes(lowerCaseInputFilter)
+            : true;
+        })
+        .map(id => {
+          const user = this.getUserInfoByAddress(id);
+          const displayName = user?.displayName || id;
+          const avatar = user?.avatar || getIdenticonAvatar(id);
+          const isCivicLiker = user?.isSubscribedCivicLiker;
+          return { id, displayName, avatar, isCivicLiker };
+        });
```
Comment (nwingt) [inline src/mixins/portfolio.js:270]: I think you should iterate with map() first, the above code seems redundant as you use most of the properties

**Comment by nwingt** [comment]
Keyword is empty [here](https://rinkeby.liker.land/zh-Hant/like1f2k6t98t9kjpdh9v9y95zhmjewuevmuwh82vj0?tab=collected)

image

File: src/components/NFTPortfolio/FilterInput.vue [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div class="flex w-full py-[10px] px-[16px] gap-[12px]">
+    <input
+      type="input"
+      class="w-full bg-transparent border-0 focus-visible:outline-none"
+      :placeholder="$t('order_menu_filter_by_input')"
+      @input="
+        (e) => {
+          $emit('handle-input-change', e.target.value);
+        }"
```
Comment (nwingt) [inline src/components/NFTPortfolio/FilterInput.vue:10]: Make code cleaner
```suggestion
      @input="$emit('handle-input-change', $event.target.value)"
```

File: src/mixins/portfolio.js [hunk 6]
```diff
@@ -246,15 +248,25 @@ export const createPortfolioMixin = ({
...
+          const user = this.getUserInfoByAddress(item.id);
+          const likerId = user?.user || item.id;
+          const lowerCaseInputFilter =
+            this.nftCreatorInputFilter?.toLowerCase() || '';
+          return lowerCaseInputFilter
+            ? item.displayName.toLowerCase().includes(lowerCaseInputFilter) ||
+                likerId.toLowerCase().includes(lowerCaseInputFilter) ||
+                item.id.includes(lowerCaseInputFilter)
+            : true;
+        });
```
Comment (nwingt) [inline src/mixins/portfolio.js:269]:
```suggestion
        .map(id => {
          const user = this.getUserInfoByAddress(id);
          const likerId = user?.user || id;
          const displayName = user?.displayName || id;
          const avatar = user?.avatar || getIdenticonAvatar(id);
          const isCivicLiker = user?.isSubscribedCivicLiker;
          return { id, likerId, displayName, avatar, isCivicLiker };
        })
        .filter(item => {
          if (!this.nftCreatorInputFilter) return true;

          const lowerCaseInputFilter =
            this.nftCreatorInputFilter.toLowerCase();
          return item.displayName.toLowerCase().includes(lowerCaseInputFilter) ||
            item.likerId.toLowerCase().includes(lowerCaseInputFilter) ||
            item.id.includes(lowerCaseInputFilter);
        });
```

**Review by nwingt — APPROVED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**